src/CalcViewModel/DataLoaders/CurrencyDataLoader.cpp

```diff
@@ -644,6 +644,18 @@ void CurrencyDataLoader::GuaranteeSelectedUnits()
             isConversionTargetSet = true;
         }
     }
+
+    // If still not set for either source or target, just select the first currency in the list
+
+    if (!isConversionSourceSet && !m_currencyUnits.empty())
+    {
+        m_currencyUnits[0].isConversionSource = true;
+    }
+
+    if (!isConversionTargetSet && !m_currencyUnits.empty())
+    {
+        m_currencyUnits[0].isConversionTarget = true;
```

Comment:
Seems like you could just have this immediately after `m_currencyUnits[0].isConversionSource = true;` -- no reason to have two identical if conditions.  I would also set `isConversionTargetSet = true;` for completeness/consistency in case something else gets added to this function later on.  Do we already have a separate issue tracking this problem?

Reply:
They are not identical, one is setting the source and the other is setting the target. I factored out the check for m_currrencyUnits.empty() though.

Reply:
Right you are.  I read too fast.